contributors/design-proposals/priority-eviction.md

```diff
@@ -0,0 +1,60 @@
+
+### By priority, then usage over requests, but only evicting pods where usage > requests
+This solution is similar in practice to the "By QoS, priority, then usage over requests" proposal, but preserves the current invariant that pods are guaranteed to be able to consume their requests without facing eviction.  Like the "By QoS, priority, then usage over requests" proposal, it exempts guaranteed pods from eviction, and thus lowers the potential for abuse by high priority pods.  Users can easily understand that their pods are evicted because they exceed their requests.  This solution provides additional availability for high-priority pods by giving priority access to remaining allocatable memory on the node and memory other pods request, but do not use. 
+
+### By Priority, QoS, then usage over requests
+This solution allows high-priority pods to consume up to their limits with very little chance of eviction, unless other high-priority pods are also present on the same node.  This solution would require using Quota controls on pod limits, rather than requests, as high-priority pods' requests make only a minor difference in their chance for eviction, and limits are a better indicator of what they can consume.  Evictions are easy for users to understand, as they can reason that a higher priority pod bursted, but does not provide a course of action to prevent evictions other than raising the priority of their own pod.
+
+### Function(priority, QoS, usage over requests)
+This solution specifies a mapping between priority, QoS, and usage - request.  For example, a possible implementation could specify that 100 points priority, 1 QoS level, and 100Mb over requests are equivalent, and then rank pods based on their "score".  It has the potential to be a solution that can balance prioritizing high-priority pods, and preventing abuse from high-priority pods.  However, it would require cluster administrators to understand this mapping in order to correctly specify priority levels, and would be prone to configuration errors.  Users would have little insight into why their pods are evicted.
```

Comment:
For the sake of evaluation, did you have a specific function in mind, or a class of functions (e.g. priority/usage, priority - usage, etc)?
I think that in all options, there would be a resource economy for cluster resources.  In some cases (e.g. #2, usage < request, priority, usage - request), this resource economy would be primarily based around request quota, and priority would play a less prominent role.  In others, (e.g. #3 priority, QoS, usage-request), a resource economy would be entirely based around priority.  In this case, it would be a combination of both priority and request quota.